### PR TITLE
Add basic ExemplarFilter doc.

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: docs/metrics/extending-the-sdk/README.md [hunk 1]
```diff
@@ -70,7 +71,48 @@ to the `MeterProvider` as shown in the example [here](./Program.cs).
...
+  and implement the `ShouldSample` method.
+
+One example is a filter, which filters all measurements of value lower
+than given threshold is given below. Such a filter prevents any measurements
+below the given threshold from ever becoming a `Exemplar`. Such filters could
+also incorporate the `TraceBasedExemplarFilter` condition as well, as storing
+exemplars for non-sampled traces may be undesired.
+
+```csharp
+public sealed class HighValueFilter : ExemplarFilter
```
Comment (cijothomas) [inline docs/metrics/extending-the-sdk/README.md:95]: could use a better name suggestion. :)

This is filtering out measurements of "low numerical value"...

File: docs/metrics/customizing-the-sdk/README.md [hunk 2]
```diff
@@ -447,6 +447,15 @@ using var meterProvider = Sdk.CreateMeterProviderBuilder()
 > As of today, there is no separate toggle for enable/disable Exemplar
 feature. It can be turned off by using `AlwaysOffExemplarFilter`.
 
+If the built-in `ExemplarFilter`s are not meeting the needs, one may author
+custom `ExemplarFilter` as shown
+[here](../extending-the-sdk/README.md#exemplarfilter). A custom filter, which
+eliminates all un-interesting measurements from becoming Exemplar is a
+recommended way to avoid performance overhead associated with collecting
```
Comment (cijothomas) [inline docs/metrics/customizing-the-sdk/README.md:454]:
```suggestion
recommended way to control performance overhead associated with collecting
```

File: docs/metrics/extending-the-sdk/README.md [hunk 3]
```diff
@@ -70,7 +71,48 @@ to the `MeterProvider` as shown in the example [here](./Program.cs).
...
+
+OpenTelemetry .NET SDK has provided the following built-in `ExemplarFilter`s:
+
+* [AlwaysOnExemplarFilter](../../../src/OpenTelemetry/Metrics/Exemplar/AlwaysOnExemplarFilter.cs)
+* [AlwaysOffExemplarFilter](../../../src/OpenTelemetry/Metrics/Exemplar/AlwaysOffExemplarFilter.cs)
+* [TraceBasedExemplarFilter](../../../src/OpenTelemetry/Metrics/Exemplar/TraceBasedExemplarFilter.cs)
+
+Custom exemplar filters can be implemented to achieve filtering based on other criterion:
+
+* `ExemplarFilter` should derive from `OpenTelemetry.ExemplarFilter>` (which
```
Comment (utpilla) [inline docs/metrics/extending-the-sdk/README.md:84]:
```suggestion
* `ExemplarFilter` should derive from `OpenTelemetry.ExemplarFilter` (which
```

File: docs/metrics/extending-the-sdk/README.md [hunk 4]
```diff
@@ -70,7 +71,48 @@ to the `MeterProvider` as shown in the example [here](./Program.cs).
...
+One example is a filter, which filters all measurements of value lower
+than given threshold is given below. Such a filter prevents any measurements
+below the given threshold from ever becoming a `Exemplar`. Such filters could
+also incorporate the `TraceBasedExemplarFilter` condition as well, as storing
+exemplars for non-sampled traces may be undesired.
+
+```csharp
+public sealed class HighValueFilter : ExemplarFilter
+{
+    private double maxValue;
```
Comment (utpilla) [inline docs/metrics/extending-the-sdk/README.md:97]:
```suggestion
    private readonly double maxValue;
```

**Review by utpilla — APPROVED** [review]

**Pull request merged**